src/main/java/space/obminyashka/items_exchange/service/impl/UserServiceImpl.java

```diff
@@ -192,7 +194,7 @@ private boolean isDurationMoreThanNumberOfDaysToKeepDeletedUser(User user) {
 
     @Override
     public void makeAccountActiveAgain(User user) {
-        user.setStatus(ACTIVE);
+        roleService.getRole(savedUsersRole.get(user.getUsername())).ifPresent(user::setRole);
```

Comment:
I think You should set only role ROLE_USER without using map for activating account, but you should check that user has role ROLE_SELF_REMOVING and after that if it is true, activate account by changing role to USER_ROLE

Reply:
or using @PreAuthorize on method in controller for checking role